messaging/src/main/java/org/axonframework/serialization/json/JacksonSerializer.java

```diff
@@ -370,7 +370,8 @@ public JacksonSerializer build() {
                 objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
             }
             if (defaultTyping) {
-                objectMapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_CONCRETE_AND_ARRAYS);
+                objectMapper.activateDefaultTyping(objectMapper.getPolymorphicTypeValidator(),
```

Comment:
The `ObjectMapper#enableDefaultTyping` itself invoked `ObjectMapper#activateDefaultTyping` using the `ObjectMapper's` `PolymoprhicTypeValidator`. Hence why I took a similar approach.